src/DTO/Address.php

```diff
@@ -6,6 +6,24 @@
 
 use JsonSerializable;
 
+/**
+ * @psalm-type TypeAddress = array{
+ *     address1: string,
```

Comment:
I think you are missing the `?` for optional array properties.
https://psalm.dev/docs/annotating_code/type_syntax/array_types/#array-shapes

A DTO can be constructed without any of these properties, there may be some edge cases.